packages/ui-number-input/src/NumberInput/styles.js

```diff
@@ -0,0 +1,166 @@
+      }
+    : {}
+  const invalidContainerStyles = invalid
+    ? {
+        borderColor: componentTheme.errorBorderColor
+      }
+    : {}
+  return {
+    numberInput: {
+      label: 'numberInput'
+    },
+    arrowContainer: {
+      label: 'numberInput_arrowContainer',
+      flex: `0 0 ${componentTheme.arrowsContainerWidth}`,
+      display: 'flex',
+      flexDirection: 'column',
+      ...disabledStyles
+    },
+    arrow: {
+      label: 'numberInput_arrow',
+      cursor: 'pointer',
+      userSelect: 'none',
+      textAlign: 'center',
+      flex: 1,
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      backgroundColor: componentTheme.arrowsBackgroundColor,
+      borderTop: 'none',
+      borderInlineEnd: 'none',
+      borderInlineStart: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.arrowsBorderColor}`,
+      borderBottom: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.arrowsBorderColor}`,
+      color: componentTheme.arrowsColor,
+      '&:last-child': { borderBottom: 'none' },
+      '&:hover': { backgroundColor: componentTheme.arrowsHoverBackgroundColor },
+      '&:active': { boxShadow: componentTheme.arrowsActiveBoxShadow }
+    },
+    inputWidth: {
+      label: 'numberInput_inputWidth',
+      display: 'block',
+      position: 'relative',
+      '&::before': {
+        content: '""',
+        pointerEvents: 'none',
+        boxSizing: 'border-box',
+        display: 'block',
+        position: 'absolute',
+        top: '-0.25rem',
+        bottom: '-0.25rem',
+        left: '-0.25rem',
+        right: '-0.25rem',
+        border: `${componentTheme.focusOutlineWidth} ${componentTheme.focusOutlineStyle} ${componentTheme.focusOutlineColor}`,
+        borderRadius: `calc(${componentTheme.borderRadius} * 1.5)`,
+        transition: 'all 0.2s',
+        opacity: 0,
+        transform: 'scale(0.95)',
+        ...focusStyles,
+        ...invalidStyles
+      }
+    },
+    inputContainer: {
+      label: 'numberInput_inputContainer',
+      display: 'flex',
+      margin: '0',
+      boxSizing: 'border-box',
+      transition: 'all 0.2s',
+      overflow: 'hidden',
+      fontFamily: componentTheme.fontFamily,
+      border: `${componentTheme.borderWidth} ${componentTheme.borderStyle} ${componentTheme.borderColor}`,
+      borderRadius: componentTheme.borderRadius,
+      ...disabledStyles,
+      ...invalidContainerStyles,
+      ...(size === 'medium'
+        ? {
+            fontSize: componentTheme.mediumFontSize,
+            height: componentTheme.mediumHeight
+          }
+        : {
+            fontSize: componentTheme.largeFontSize,
+            height: componentTheme.largeHeight
+          })
+    },
+    input: {
+      label: 'numberInput_input',
+      all: 'initial',
+      textAlign: 'start',
+      direction: 'inherit',
+      '&::-ms-clear': { display: 'none' },
```

Comment:
this is no longer needed in the browsers we support, see https://developer.mozilla.org/en-US/docs/Archive/Web/CSS/::-ms-clear